### PR TITLE
refactor: Post deploy network updates for RHEL

### PR DESCRIPTION
**Pull request by rayjh**
Remove writing of interface configuration files during gather mac
addresses. Force ipv6 link local neighbor discovery using ip command.

Add fixes to RHEL interface Jinja2 templates to convert 'true/false'
to 'yes/no' and handle quotes in 'BONDING_OPTS'.

Add backup of t0 interface configuration files for RHEL.
Move backup of Ubuntu t0 interfaces file to /etc/network/gen-t0-bkup.

### CODE REVIEW

File: playbooks/activate_client_interfaces.yml [hunk 1]
```diff
@@ -32,48 +33,78 @@
            item.replace('-', '_')]['active'] }}"
       with_items: "{{ ansible_interfaces }}"
 
-    # A one time copy is made of the network interfaces files after initial
-    # OS install. This is now attempted at the end of install in wait_for_
-    # client_ping.yml, but is re-attempted here for previous installs done
-    # with prior versions of Genesis.
+    - pause:
```
Comment (jaywcarman) [inline playbooks/activate_client_interfaces.yml]: [ANSIBLE0011] All tasks should be named

File: playbooks/activate_client_interfaces.yml [hunk 2]
```diff
@@ -32,48 +33,78 @@
...
+      when:
+        - "{{ item != 'lo' }}"
+        - "{{ ansible_host != hostvars[inventory_hostname]['ansible_' + \
+           item.replace('-', '_')].get('ipv4', {}).get('address') }}"
+      with_items: "{{ ansible_interfaces }}"
+      ignore_errors: yes
 
-        - name: Backup interfaces file
-          command: "cp /etc/network/interfaces /etc/network/interfaces.bak"
+    - pause:
```
Comment (jaywcarman) [inline playbooks/activate_client_interfaces.yml]: [ANSIBLE0011] All tasks should be named

File: playbooks/activate_client_interfaces.yml [hunk 3]
```diff
@@ -32,48 +33,78 @@
...
+      when:
+        - "{{ item != 'lo' }}"
+        - "{{ ansible_host != hostvars[inventory_hostname]['ansible_' + \
+           item.replace('-', '_')].get('ipv4', {}).get('address') }}"
       with_items: "{{ ansible_interfaces }}"
+      ignore_errors: yes
 
-    - name: Bring up all interfaces
-      command: "ip link set dev {{ item }} up"
+    - pause:
```
Comment (jaywcarman) [inline playbooks/activate_client_interfaces.yml]: [ANSIBLE0011] All tasks should be named

File: playbooks/activate_client_interfaces.yml [hunk 4]
```diff
@@ -32,48 +33,78 @@
...
-        owner: "root"
-        group: "root"
-        mode: "0644"
-      when: (ansible_distribution == 'CentOS') or
-            (ansible_distribution == 'RedHat')
+    # Update switch mac address tables by adding temporary link local
+    # ipv4 address.  This also forces an ipv6 link local setup which
+    # drives traffic to the switch.
+    - name: Update switch mac address tables
+      command: "sudo ip addr add 169.254.0.2/24 dev {{ item }}"
```
Comment (jaywcarman) [inline playbooks/activate_client_interfaces.yml]: I'm not sure if there's any _real_ benefit, but the preferred way elevate privilege in Ansible is to use "become". e.g.: `become: true`

File: playbooks/activate_client_interfaces.yml [hunk 5]
```diff
@@ -32,48 +33,78 @@
...
       with_items: "{{ ansible_interfaces }}"
+      ignore_errors: yes
 
-    - name: Bring up all interfaces
-      command: "ip link set dev {{ item }} up"
+    - pause:
+        seconds: 5
+
+    - name: clear the temporary addresses
+      command: "sudo ip addr del 169.254.0.2/24 dev {{ item }}"
```
Comment (jaywcarman) [inline playbooks/activate_client_interfaces.yml]: ditto on `become: true`

File: playbooks/activate_client_interfaces.yml (hunk 4 above)
Comment (jaywcarman) [reply playbooks/activate_client_interfaces.yml]: Also, we're actually running as root on the client nodes anyways, so you shouldn't even need to escalate privilege.

File: playbooks/activate_client_interfaces.yml (hunk 5 above)
Comment (rayjh) [reply playbooks/activate_client_interfaces.yml]: done

File: playbooks/activate_client_interfaces.yml (hunk 3 above)
Comment (rayjh) [reply playbooks/activate_client_interfaces.yml]: done

File: playbooks/activate_client_interfaces.yml (hunk 4 above)
Comment (rayjh) [reply playbooks/activate_client_interfaces.yml]: changed to become: true

File: playbooks/activate_client_interfaces.yml (hunk 2 above)
Comment (rayjh) [reply playbooks/activate_client_interfaces.yml]: done

File: playbooks/activate_client_interfaces.yml (hunk 1 above)
Comment (rayjh) [reply playbooks/activate_client_interfaces.yml]: done

**Pull request merged**